algolia-search/src/algolia-search.island.tsx

```diff
@@ -0,0 +1,111 @@
+import algoliasearch from 'algoliasearch/lite';
+import {createIslandWebComponent} from 'preact-island'
+import {HitsProps, InstantSearch, useHits} from 'react-instantsearch';
+import SearchForm from "./search-form";
+import EventHit from "./hits/events";
+import NewsHit from "./hits/news";
+import DefaultHit from "./hits/default-hit";
+import styled from "styled-components";
+import {StanfordHit} from "./hits/hit.types";
+
+const islandName = 'algolia-search'
+const appId = window.drupalSettings?.stanfordAlgolia.appId || process.env.ALGOLIA_APP_ID
+const key = window.drupalSettings?.stanfordAlgolia.searchKey || process.env.ALGOLIA_KEY
+const searchClient = algoliasearch(appId, key);
+
+const Hit = ({hit}: HitsProps<StanfordHit>) => {
+  if (hit.type === 'Event') return <EventHit hit={hit}/>
+  if (hit.type === 'News') return <NewsHit hit={hit}/>
+
+  return <DefaultHit hit={hit}/>
+}
+
+const Container = styled.div`
+
+  li {
+    margin-bottom: 10px;
+
+    &:last-child {
+      border-bottom: none;
+    }
+  }
+
+  [type=checkbox],[type=radio] {
+    -webkit-clip-path: unset;
+    padding: 0;
+    width:  12px;
+    height:12px;
+    clip: unset;
+    overflow: unset;
+    position: relative;
+    clipPath: unset;
+    margin-right: 8px;
+  }
+  fieldset {
+    padding: 0;
+  }
+`
+
+const CustomHits = (props) => {
+  const {hits} = useHits(props);
+  if (hits.length === 0) return (
+    <p>No results for your search. Please try another search.</p>
+  )
+
+  // Returns results
+  return (
+    <ul
+      style={{listStyle: "none", padding: 0, float: "right", width: "66%"}}
+      className=""
+    >
+      {hits.map(hit =>
+        <li key={hit.objectID}>
+          <Hit hit={hit}/>
+        </li>
+      )}
+    </ul>
+  )
+}
+
+
+const Search = () => {
+  const currentSearchParams = new URLSearchParams(window.location.search)
+
+  const initialUiState = {};
+
+  if (currentSearchParams.get('key')) {
+    initialUiState.query = currentSearchParams.get('key');
+  }
+  if (currentSearchParams.get("page-type")) {
+    initialUiState.refinementList = {basic_page_type: currentSearchParams.get("page-type").split(',')}
+  }
+  if (currentSearchParams.get("shared")) {
+    initialUiState.refinementList = {basic_page_type: currentSearchParams.get("shared").split(',')}
```

Comment:
the key needs to match the key for the facets. in your case it will be `shared_tags` instead of `basic_page_type`
the GET parameter also needs to match what you set in the `useEffect` hook